### ISSUE TITLE
Petstore fails with "collection format Multi is not supported"

### ISSUE THREAD

**Issue by scott-lin** [Azure/go-autorest#309]
I'm running [petstore](http://petstore.swagger.io/v2/swagger.json) through AutoRest Go, and it's failing with the below error. I would have expected this widely used example to just work.

```
CMD> autorest --go --input-file=petstore.txt --output-folder=petstore-go

AutoRest code generation utility [version: 2.0.4283; node: v8.9.1]
(C) 2018 Microsoft Corporation.

https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\scottlin.REDMOND\.autorest\@microsoft.azure_autorest-core@2.0.4283\node_modules\@microsoft.azure\autorest-core\dist' (2.0.4283)
   Loading AutoRest extension '@microsoft.azure/autorest.go' (~2.1.47->2.1.110)
   Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.38->2.3.38)
   
FATAL: System.NotSupportedException: Collection format Multi is not supported.
   at AutoRest.Go.Extensions.GetSeparator(CollectionFormat format) in /opt/vsts/work/1/s/src/Extensions.cs:line 251
   at AutoRest.Go.Model.ParameterGo.ValueForMap(Boolean useDefault) in /opt/vsts/work/1/s/src/Model/ParameterGo.cs:line 171
   at AutoRest.Go.Model.ParameterGoExtensions.<>cDisplayClass0_0.<BuildParameterMap>b2(ParameterGo p) in /opt/vsts/work/1/s/src/Model/ParameterGo.cs:line 306
   at AutoRest.Core.Utilities.Extensions.ForEach[T](IEnumerable1 sequence, Action1 action) in /opt/vsts/work/1/s/autorest.common/src/Utilities/Extensions.cs:line 47
   at AutoRest.Go.Model.ParameterGoExtensions.BuildParameterMap(IEnumerable`1 parameters, String mapVariable) in /opt/vsts/work/1/s/src/Model/ParameterGo.cs:line 303
   at AutoRest.Go.Model.MethodGo.get_QueryMap() in /opt/vsts/work/1/s/src/Model/MethodGo.cs:line 311
   at AutoRest.Go.Templates.MethodTemplate.<ExecuteAsync>d__1.MoveNext() in /opt/vsts/work/1/s/src/obj/Razor/Templates/MethodTemplate.cshtml:line 134
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AutoRest.Core.Template`1.ToString() in /opt/vsts/work/1/s/autorest.common/src/Template.cs:line 217
   at AutoRest.Core.Template`1.Include[TU,TV](TU template, TV templateModel) in /opt/vsts/work/1/s/autorest.common/src/Template.cs:line 147
   at AutoRest.Go.Templates.ServiceClientTemplate.<ExecuteAsync>d__1.MoveNext() in /opt/vsts/work/1/s/src/obj/Razor/Templates/ServiceClientTemplate.cshtml:line 117
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AutoRest.Core.CodeGenerator.<Write>d__12.MoveNext() in /opt/vsts/work/1/s/autorest.common/src/CodeGenerator.cs:line 119
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AutoRest.Go.CodeGeneratorGo.<Generate>d__6.MoveNext() in /opt/vsts/work/1/s/src/CodeGeneratorGo.cs:line 64
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AutoRest.Go.Program.<ProcessInternal>d__3.MoveNext() in /opt/vsts/work/1/s/src/Program.cs:line 107
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at NewPlugin.<Process>d__20.MoveNext() in /opt/vsts/work/1/s/autorest.common/src/Plugins/NewPlugin.cs:line 163
FATAL: go/generate - FAILED
FATAL: Error: Plugin go reported failure.
Process() cancelled due to exception : Plugin go reported failure.
  Error: Plugin go reported failure.
```

**Comment by vladbarosan** [comment]
Hi @scott-lin  As of now we dont support this, is there a specific API you are using it that needs this feature ?

**Comment by scott-lin** [comment]
No, I do not have a specific API. I was simply investigating AutoRest + Go, and I ran the Petstore sample though it to try it out. Thought I'd report the error since others might try the same.

**Comment by jhendrixMSFT** [comment]
This has been fixed in generator v2.1.137
